lib/req/steps.ex

```diff
@@ -1336,6 +1342,19 @@ defmodule Req.Steps do
     decompress_body(rest, :zlib.gunzip(body), acc)
   end
 
+  defp decompress_body(["deflate" | rest], body, acc) do
+    # uncompress/1 is used when the body *has* zlib headers and checksum
+    # unzip/1 works when the body has no headers or checksum.
+    unzipped =
+      try do
+        :zlib.uncompress(body)
+      rescue
+        ErlangError -> :zlib.unzip(body)
+      end
```

Comment:
Someone with greater knowledge of the deflate representation could probably pattern match on the start of the `body` binary to determine if it had the header and checksum, but that's beyond me... 😬